build-logic/convention/src/main/kotlin/AndroidApplicationConventionPlugin.kt

```diff
@@ -0,0 +1,122 @@
+import com.android.build.api.dsl.ApplicationExtension
+import io.homeassistant.companion.android.getPluginId
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.kotlin.dsl.apply
+import org.gradle.kotlin.dsl.configure
+import org.gradle.kotlin.dsl.withType
+import org.jetbrains.kotlin.gradle.dsl.JvmTarget
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
+private const val APPLICATION_ID = "io.homeassistant.companion.android"
+
+/**
+ * A convention plugin that applies common configurations to Android application modules.
+ * This avoid duplicating the same configuration across multiple modules.
+ *
+ * It applies multiple gradle plugins that are used in every modules.
+ *
+ * Once this plugin is applied, the values set can be overridden if needed. But it probably
+ * means that it should not in this convention plugin anymore.
+ *
+ * application `version code` can be set in the environment variable VERSION_CODE
+ * keystore information can be set in the environment variables:
+ * - KEYSTORE_PATH for the path to the keystore
+ * - KEYSTORE_PASSWORD for the store password
+ * - KEYSTORE_ALIAS for the alias
+ * - KEYSTORE_ALIAS_PASSWORD for the alias password
+ */
+class AndroidApplicationConventionPlugin : Plugin<Project> {
```

Comment:
I didn't give a lot of though to this class for now it's mostly a copy paste. But in the future we might want to split this into smaller plugins more specifics. Since pieces of this can be shared with `:common`.

Reply:
+ some pieces can be shared with `wear`

Reply:
My comment was wrong about wear and I deleted it before your reply but apparently not fast enough :)